### PR TITLE
NUI-1644: Add parallel block transfer for upload/download to SDK

### PR DESCRIPTION
**Pull request by jdelbick**
## Description

Use the newest version of `node-httptransfer` with block transfer

- replace `downloadFile` with `downloadFileConcurrently`
   - a few slight differences:
   - pass through `fileSize` and `contentType` to avoid head request
   - get request must return `content-length` header (had to add this to mocks)
       - NOTE: we should move to mocking `node-httptransfer` methods instead of mocking the http request directly to avoid the issue of tests breaking and start testing invidual, isolated components in a predictable way
- replace `uploadFile` with `uploadFileConcurrently`
- replace `uploadAEMMultipartFile` with `uploadMultiPartFileConcurrently`

Fixes # https://jira.corp.adobe.com/browse/NUI-1644

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by pheenomenon — APPROVED** [review]

File: lib/storage/http.js [hunk 1]
```diff
@@ -21,10 +21,13 @@ const MAX_RETRY_DURATION_UPLOAD = 900000; // 15 mins
...
         console.log(`downloading asset ${AssetComputeLogUtils.redactUrl(asset.url)} into ${file}\nheaders:`, asset.headers);
+        console.log(`asset content type: ${asset.type}, size: ${asset.size}`);
 
-        await http.downloadFile(asset.url, file, {
+        await http.downloadFileConcurrently(asset.url, file, {
             retryEnabled: !process.env.ASSET_COMPUTE_DISABLE_RETRIES,
-            headers: asset.headers
+            headers: asset.headers,
+            contentType: asset.type || asset.mimeType,
+            fileSize: asset.size
```
[the remark this answers is not in the document]
Comment (pheenomenon) [reply lib/storage/http.js:30]: wonder if we need a test where it does HEAD request?

File: test/engine.test.js [hunk 2]
```diff
@@ -914,7 +918,7 @@ describe("Pipeline Engine tests", function () {
         // assert download ran
         assert(nock.isDone());
         await pipeline.run(plan);
-    });
+    }).timeout(20000);
```
[the remark this answers is not in the document]
Comment (jdelbick) [reply test/engine.test.js:921]: yep, nice catch :)

File: lib/storage/http.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdelbick) [reply lib/storage/http.js:30]: this test does a head req: https://github.com/adobe/asset-compute-pipeline/blob/313614702f0bcefc0f3144943aa6eb09ed2ed79e/test/storage/http.test.js#L67-L93

**Review by bobvanmanen — APPROVED** [review]
LGTM

**Pull request merged**